### PR TITLE
10489: Status Order Report Enhancements

### PR DESCRIPTION
**Pull request by nechama-krigsman**
flexion#10489

### CODE REVIEW

**Review by btejha — APPROVED** [review]

File: web-client/src/presenter/actions/CourtIssuedDocketEntry/setCourtIssuedDocumentInitialDataAction.test.ts [hunk 1]
```diff
@@ -108,4 +122,50 @@ describe('setCourtIssuedDocumentInitialDataAction', () => {
...
+        docketEntryId: docketEntryIds[2],
+      },
+      state: {
+        caseDetail: MOCK_CASE,
+        form: {},
+        judges: [
+          {
+            judgeFullName: 'John O. Colvin',
+            name: 'Colvin',
+            userId: '4497d22-0874-4c5e-ac98-a843d1472baf',
```
Comment (Mwindo) [inline web-client/src/presenter/actions/CourtIssuedDocketEntry/setCourtIssuedDocumentInitialDataAction.test.ts]: Here and below, can we get this data from the `judgeColvin` user in `mockUsers.ts`?

File: web-api/src/business/useCases/courtIssuedOrder/fileCourtIssuedOrderInteractor.ts [hunk 2]
```diff
@@ -40,11 +47,22 @@ export const fileCourtIssuedOrder = async (
...
+    documentMetadata.jurisdiction === 'retained'
+  ) {
+    const ojrEventCode = COURT_ISSUED_EVENT_CODES.find(
+      e => e.eventCode === 'OJR',
+    );
+    documentMetadata.documentType = ojrEventCode?.documentType;
+    documentMetadata.eventCode = 'OJR';
+  }
+
+  if (['O', 'NOT', 'OJR'].includes(documentMetadata.eventCode)) {
```
Comment (Mwindo) [inline web-api/src/business/useCases/courtIssuedOrder/fileCourtIssuedOrderInteractor.ts]: Would it be worth pulling these event codes out into a self-documenting constant like `eventCodesThatAllowFreeText`? (There is probably a better name, but you get the idea!)

File: web-api/src/business/useCases/courtIssuedOrder/fileCourtIssuedOrderInteractor.test.ts [hunk 3]
```diff
@@ -507,4 +473,334 @@ describe('fileCourtIssuedOrderInteractor', () => {
...
+    describe('eventCode "O"', () => {
+      it('should add order document to case and set freeText and draftOrderState.freeText correctly for orderType status report', async () => {
+        await fileCourtIssuedOrderInteractor(
+          applicationContext,
+          {
+            documentMetadata: {
+              draftOrderState: {},
+              dueDate: '2024-11-05',
+              eventCode: 'O',
+              orderType: 'statusReport',
```
Comment (Mwindo) [inline web-api/src/business/useCases/courtIssuedOrder/fileCourtIssuedOrderInteractor.test.ts]: Can we use constants when possible in this file? E.g., here: `STATUS_REPORT_ORDER_OPTIONS.orderTypeOptions.statusReport`

Comment (Mwindo) [reply web-api/src/business/useCases/courtIssuedOrder/fileCourtIssuedOrderInteractor.test.ts]: Anywhere we have 'statusReport', 'statusReportStipulatedDecision', 'retained', or 'restoredToGeneralDocket', we should be able to use the `STATUS_REPORT_ORDER_OPTIONS` object.

File: cypress/local-only/tests/integration/statusReportOrder/status-report-order-description-fields.cy.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+describe('should default status report order descriptions', () => {
+  const today = formatNow(FORMATS.MMDDYYYY);
+  it('should display default description when document type is an Order', () => {
+    judgeOrChambersCreatesStatusReportOrder(today);
+    loginAsDocketClerk();
+    cy.visit(`/case-detail/${docketNumber}`);
+    cy.get('#tab-drafts').click();
+    getLastDraftOrderElementFromDrafts().click();
+    cy.get('[data-testid="add-court-issued-docket-entry-button"]').click();
+    cy.get('.select-react-element__control').should('have.text', 'Order');
```
Comment (Mwindo) [inline cypress/local-only/tests/integration/statusReportOrder/status-report-order-description-fields.cy.ts]: Let's try avoiding third-party library selectors. I think we can just do `#document-type` here.

File: cypress/local-only/tests/integration/statusReportOrder/status-report-order-description-fields.cy.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  });
+
+  it('should set event code to OJR when case is stricken from trial session and jurisdiction is retained and display default description', () => {
+    judgeOrChambersCreatesStatusReportOrder(today, true);
+    loginAsDocketClerk();
+    cy.visit(`/case-detail/${docketNumber}`);
+    cy.get('#tab-drafts').click();
+    getLastDraftOrderElementFromDrafts().click();
+    cy.get('[data-testid="add-court-issued-docket-entry-button"]').click();
+    cy.get('.select-react-element__control').should(
```
Comment (Mwindo) [inline cypress/local-only/tests/integration/statusReportOrder/status-report-order-description-fields.cy.ts:43]: See comment above

File: cypress/local-only/tests/integration/statusReportOrder/status-report-order-description-fields.cy.ts (hunk 4 above)
Comment (Mwindo) [reply cypress/local-only/tests/integration/statusReportOrder/status-report-order-description-fields.cy.ts]: Same thing anyplace we use this selector below.

**Review by Mwindo — APPROVED** [review]
I left a few comments upon which my approval is contingent, but generally looks good! Thanks @nechama-krigsman and @btejha!

**Review by TomElliottFlexion — APPROVED** [review]

**Pull request merged**